lib/tasks/clean_up_orphaned_fixity_records.rake

```diff
@@ -0,0 +1,61 @@
+# Context:
+# When a fileset is deleted from Fedora, the records for its fixity checks remain in the database.
+# This creates confusion when attempting to ensure that all current files have had their fixity
+# checked recently.
+#
+# What this job does:
+# Loop through all fixity records and verify that they are associated with a fileset that we
+# still have on hand. If it turns out they are not, delete the "orphaned" fixity check record.
+
+namespace :chf do
+  desc "run fixity checks with logs and notification on failure"
+  task :clean_up_orphaned_fixity_checks => :environment do
+    @conn = Blacklight.default_index.connection
+    x = 0
+    n = 100
+    loop do
+      batch = ChecksumAuditLog.select(:file_set_id).distinct[x..x+n-1]
```

Comment:
Not actually sure what this is doing, but I don't think it's something good.  You're trying to fetch all the rows, but in batches to avoid having them all in memory at once?  Good plan. 

The right way to loop through all rows in an rdbms with ActiveRecord would be using the `limit` and `offset` methods. http://guides.rubyonrails.org/active_record_querying.html#limit-and-offset

However, I think we can do one better, and just use ActiveRecord's find_in_batches to do it for us. 

I am not sure if you are trying to loop through _all_ the rows, or just the ones representing most recent checks?

```ruby
ChecksumAuditLog.find_in_batches(batch_size: 100) do |array_of_one_hundred|
# or perhaps
ChecksumAuditLog.latest_checks.find_in_batches(batch_size: 100) do |array_of_one_hundred|
```
Alternately to find_in_batches, if it is more convenient to yield each row individually, but still fetch in batches:

```ruby
ChecksumAuditLog.find_each(batch_size: 100) do |checksum_audit_log_record|
ChecksumAuditLog.latest_checks.find_each(batch_size: 100) do |checksum_audit_log_record|
```
http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each

The default batch size of 1000 _may_ be just fine in either case, but 100 is fine too. 

Having trouble understanding the rest of the code, how about you start refactoring to use one of these approaches, and then I can re-review?

(In general, if you find yourself using `loop do` in ruby, there's _probably_ a better way).

Reply:
(Oh, I see, it's 100 becuase you may be checking them all in solr at once? ok, 100 is fine!)